src/Features/Core/Portable/InvertConditional/AbstractInvertConditionalCodeRefactoringProvider.cs

```diff
@@ -31,20 +34,44 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
             }
 
             context.RegisterRefactoring(new MyCodeAction(
-                c => InvertConditionalAsync(document, span, c)),
+                c => InvertConditionalAsync(document, conditional, c)),
                 conditional.Span);
         }
 
         private static async Task<TConditionalExpressionSyntax?> FindConditionalAsync(
             Document document, TextSpan span, CancellationToken cancellationToken)
             => await document.TryGetRelevantNodeAsync<TConditionalExpressionSyntax>(span, cancellationToken).ConfigureAwait(false);
 
-        private static async Task<Document> InvertConditionalAsync(
-            Document document, TextSpan span, CancellationToken cancellationToken)
+        protected override async Task FixAllAsync(Document document, ImmutableArray<TextSpan> fixAllSpans, SyntaxEditor editor, CancellationToken cancellationToken)
         {
-            var conditional = await FindConditionalAsync(document, span, cancellationToken).ConfigureAwait(false);
-            Contract.ThrowIfNull(conditional);
+            var originalRoot = editor.OriginalRoot;
+
+            // Get all conditional nodes in the given fixAllSpans.
+            var conditionals = originalRoot.DescendantNodes().OfType<TConditionalExpressionSyntax>()
+                .Where(node => fixAllSpans.Any(fixAllSpan => fixAllSpan.IntersectsWith(node.Span)));
+
+            // We're going to be continually editing this tree. Track all the nodes we
+            // care about so we can find them across each edit.
+            document = document.WithSyntaxRoot(originalRoot.TrackNodes(conditionals));
+
+            foreach (var originalConditional in conditionals.Reverse())
+            {
+                // Only process conditionals fully within fixAllSpan
+                if (!fixAllSpans.Any(fixAllSpan => fixAllSpan.Contains(originalConditional.Span)))
+                    continue;
 
+                var currentRoot = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                var currentConditional = currentRoot.GetCurrentNodes(originalConditional).Single();
+                document = await InvertConditionalAsync(document, currentConditional, cancellationToken).ConfigureAwait(false);
```

Comment:
nits:
1. going back to document seems unneessary.
2. we have a SyntaxEditor, so we likely can just do normal .ReaplceNode calls.  Using the right overload means we'll also invert properly when a nested statement is inverted in a containing if-statement.
--

i'm also just on the fence here because even thinking about it, this feature makes no sense for fix-all :D

Reply:
> going back to document seems unneessary.

I guess this came from us re-using the same method for both the original code action and inside the loop within the FixAllAsync override.

> i'm also just on the fence here because even thinking about it, this feature makes no sense for fix-all :D

Let me file an issue to revert the FixAll support for this refactoring once we have FixAll for another one, with appropriate replacement unit tests and integration tests. I'd like to avoid spending another cycle for doing this as part of this PR and block us from adding this feature and enabling FixAll support for other refactorings.

Reply:
Filed #60956 for the revert